shap/__init__.py

```diff
@@ -73,13 +74,56 @@ def __getattribute__(self, item):
 
 
 # other stuff :)
-from . import datasets
-from . import utils
-from . import links
-
+from . import datasets, links, utils
 from .actions._optimizer import ActionOptimizer
+from .utils import approximate_interactions, sample
 
 #from . import benchmark
-
 from .utils._legacy import kmeans
-from .utils import sample, approximate_interactions
+
+# Use __all__ to let type checkers know what is part of the public API.
+__all__ = [
+    # Explanations
+    "Cohorts",
+    "Explanation",
+    "AdditiveExplainer",
+    "DeepExplainer",
+    "Explainer",
+    "GPUTreeExplainer",
+    "GradientExplainer",
+    "KernelExplainer",
+    "LinearExplainer",
+    "PartitionExplainer",
+    "PermutationExplainer",
+    "SamplingExplainer",
+    "TreeExplainer",
+    "other",
+
+    # Plots
+    "plots",
+    "initjs",
+    "getjs",
+    "save_html",
+    "bar_plot",
+    "decision_plot",
+    "dependence_plot",
+    "embedding_plot",
+    "force_plot",
+    "group_difference_plot",
+    "image_plot",
+    "monitoring_plot",
+    "multioutput_decision_plot",
+    "partial_dependence_plot",
+    "summary_plot",
+    "text_plot",
+    "waterfall_plot",
+
+    # Other stuff
+    "datasets",
+    "utils",
+    "links",
+    "ActionOptimizer",
+    "kmeans",
+    "sample",
+    "approximate_interactions",
```

Comment:
```suggestion
__all__ = [
    "Cohorts",
    "Explanation",

    # Explainers
    "other",
    "AdditiveExplainer",
    "DeepExplainer",
    "Explainer",
    "GPUTreeExplainer",
    "GradientExplainer",
    "KernelExplainer",
    "LinearExplainer",
    "PartitionExplainer",
    "PermutationExplainer",
    "SamplingExplainer",
    "TreeExplainer",

    # Plots
    "plots",
    "bar_plot",
    "summary_plot",
    "decision_plot",
    "multioutput_decision_plot",
    "embedding_plot",
    "force_plot",
    "getjs",
    "initjs",
    "save_html",
    "group_difference_plot",
    "image_plot",
    "monitoring_plot",
    "partial_dependence_plot",
    "dependence_plot",
    "text_plot",
    "waterfall_plot",

    # Other stuff
    "datasets",
    "links",
    "utils",
    "ActionOptimizer",
    "approximate_interactions",
    "sample",
    "kmeans",
```

Reply:
Would it be better to sort each group in `__all__` in alphabetical order? That could make it easier for users to find a particular function, for example when using autocomplete in Jupyter.

The order that `isort` chooses is pretty arbitrary from the point of view of users, as it's determined by the order of our private submodules. e.g. `dependence_plot` comes later in the list as it happens to be defined in `_scatter.py`.

Reply:
Does the ordering in Jupyter's autocomplete options depend on how we sort `__all__`?
If it does then I agree with you.

Otherwise, I think the `__all__` ordering is more for maintainers than users. It's easier to edit both the imports and the `__all__` list in tandem when they are ordered the same way.

Reply:
Ah, I tested it and I was mistaken, actually Jupyter just uses alphabetic order. I'll do as you suggest and order `__all__` in accordance with the imports.